js/app.js

```diff
@@ -0,0 +1,548 @@
+    'isoAlpha3': { name: 'isoAlpha3', mysql: 'char(3) DEFAULT NULL', firebird: 'char(3) DEFAULT NULL', checked: false },
+    'geonameId': { name: 'geonameId', mysql: 'int(10) DEFAULT NULL', firebird: 'integer DEFAULT NULL', checked: false }
+};
+
+var columnsLookup = [
+    { name: 'countryCode', mysql: 'char(2) NOT NULL DEFAULT \'\'', firebird: 'char(3) NOT NULL', checked: true },
+    { name: 'languages', mysql: 'varchar(100) DEFAULT NULL', firebird: 'varchar(100) DEFAULT NULL', checked: false }
+];
+
+var settings = {
+    'languagelookup': { name: 'languagelookup', longName: 'Language Lookup tables', checked: false, disabled: false, supportedOutputs: [OUTPUT_MYSQL, OUTPUT_FIREBIRD] }
+};
+
+var outputTypes = {};
+outputTypes[OUTPUT_MYSQL] = { name: OUTPUT_MYSQL };
+outputTypes[OUTPUT_FIREBIRD] = { name: OUTPUT_FIREBIRD };
+outputTypes[OUTPUT_XML] = { name: OUTPUT_XML };
+outputTypes[OUTPUT_JSON] = { name: OUTPUT_JSON };
+outputTypes[OUTPUT_CSV] = { name: OUTPUT_CSV };
+outputTypes[OUTPUT_YAML] = { name: OUTPUT_YAML };
+
+var GeneratorApp = React.createClass({
+    getInitialState: function() {
+        return {
+            columns: this.props.columns,
+            settings: this.props.settings,
+            outputTypes: this.props.outputTypes,
+            selectedOutputType: OUTPUT_MYSQL,
+            output: ''
+        };
+    },
+    componentWillMount: function() {
+        this.getOutput();
+    },
+    // handles changes for all the checkboxes and radio buttons
+    toggleCheck: function(objectKey, type, data) {
+        // radio buttons are handled different than checkboxes
+        if (type === 'outputTypes') {
+            var settings = this.state.settings;
+
+            // based on the selected, disable or enable 'languagelookup' setting
+            for (var key in settings) {
+                if (settings.hasOwnProperty(key) && settings[key].name === 'languagelookup') {
+                    if (settings[key].supportedOutputs.indexOf(data[objectKey].name) === -1) {
+                        settings[key].checked = false;
+                        settings[key].disabled = true;
+                    } else {
+                        settings[key].disabled = false;
+                    }
+                }
+            }
+
+            this.setState({selectedOutputType: data[objectKey].name});
+        } else if (type === 'settings') {
+            if (data[objectKey].name === 'languagelookup' && !data[objectKey].checked) {
+                var columns = this.state.columns;
+
+                columns['languages'].checked = true;
+            }
+
+            data[objectKey].checked = !data[objectKey].checked;
+        } else {
+            data[objectKey].checked = !data[objectKey].checked;
+        }
+
+        this.forceUpdate();
+
+        // TODO: do I need this? How forceUpdate does updates based on references?
+//        this.setState({type: data});
```

Comment:
Use `setState` wherever you can. `forceUpdate` should be your last exit. `setState` will trigger re-render of component and this is usually what you want.